src/mds/mdstypes.h

```diff
@@ -393,10 +395,10 @@ struct inode_t {
   utime_t    mtime;   // file data modify time.
   utime_t    atime;   // file data access time.
   uint32_t   time_warp_seq;  // count of (potential) mtime/atime timewarps (i.e., utimes())
-  bufferlist inline_data;
+  bufferlist *inline_data;
   version_t  inline_version;
 
-  std::map<client_t,client_writeable_range_t> client_ranges;  // client(s) can write to these ranges
+  compact_map<client_t,client_writeable_range_t> client_ranges;  // client(s) can write to these ranges
```

Comment:
I'm not sure about this one — don't we have a value in client_ranges for basically every file that a client opens for write?